scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -539,3 +539,103 @@ def small_sample_generator(n_dims):
                     stats._stats_py._broadcast_concatenate(samples, axis)
                 with pytest.raises(ValueError, match=message):
                     hypotest(*samples, *args, axis=axis, **kwds)
+
+
+def test_masked_array_2_sentinel_array():
+    # prepare arrays
+    np.random.seed(0)
+    A = np.random.rand(10, 11, 12)
+    B = np.random.rand(12)
+    mask = A < 0.5
+    A = np.ma.masked_array(A, mask)
+
+    # set arbitrary elements to special values
+    # (these values might have been considered for use as sentinel values)
+    max_float = np.finfo(np.float64).max
+    eps = np.finfo(np.float64).eps
+    A[3, 4, 1] = np.nan
+    A[4, 5, 2] = np.inf
+    A[5, 6, 3] = max_float
+    B[8] = np.nan
+    B[7] = np.inf
+    B[6] = max_float * (1 - 2*eps)
+
+    # convert masked A to array with sentinel value, don't modify B
+    out_arrays, sentinel = _masked_arrays_2_sentinel_arrays([A, B])
+    A_out, B_out = out_arrays
+
+    # check that good sentinel value was chosen (according to intended logic)
+    assert (sentinel != max_float) and (sentinel != max_float * (1 - 2*eps))
+    assert sentinel == max_float * (1 - 2*eps)**2
+
+    # check that output arrays are as intended
+    A_reference = A.data
+    A_reference[A.mask] = sentinel
+    np.testing.assert_array_equal(A_out, A_reference)
+    assert B_out is B
+
+
+def test_masked_stat_1d():
+    # Adding support for masked arrays requires minimal modifications to the
+    # _axis_nan_policy decorator, so exhaustive tests for masked arrays
+    # (and all possible combinations of masked/non masked arrays with and
+    # without masked elements and nans) is not necessary. Instead, perform
+    # basic tests in 1D and 2D.
+    males = [19, 22, 16, 29, 24]
+    females = [20, 11, 17, 12]
+    res = stats.mannwhitneyu(males, females)
+
+    # same result when extra nan is omitted
+    females2 = [20, 11, 17, np.nan, 12]
+    res2 = stats.mannwhitneyu(males, females2, nan_policy='omit')
+    np.testing.assert_array_equal(res2, res)
+
+    # same result when extra element is masked
+    females3 = [20, 11, 17, 1000, 12]
+    mask3 = [False, False, False, True, False]
+    females3 = np.ma.masked_array(females3, mask=mask3)
+    res3 =  stats.mannwhitneyu(males, females3)
```

Comment:
```suggestion
    res3 = stats.mannwhitneyu(males, females3)
```
PEP8: `scipy/stats/tests/test_axis_nan_policy.py:597:11: E222 multiple spaces after operator`